client/grpc/tmservice/service.go

```diff
@@ -192,16 +215,26 @@ func (s queryServer) GetNodeInfo(ctx context.Context, req *GetNodeInfoRequest) (
 	return &resp, nil
 }
 
+func (s queryServer) ABCIQuery(ctx context.Context, req *ABCIQueryRequest) (*ABCIQueryResponse, error) {
+	if s.queryFn == nil {
+		return nil, status.Error(codes.Internal, "ABCI Query handler undefined")
```

Comment:
should we add this check in `NewQueryServer`? so that it errors on startup

Reply:
Sure, that's a good idea. But that means we'll have to break the `Application` interface such that `RegisterTendermintService` now would return an error. WDYT?

Reply:
Yeah let's not break that interface. How about just panicking? It's at startup, so it's IMO okay.

Reply:
The only place to do that would be in `client/grpc/tmservice/service.go`, which doesn't seem right to me -- it should error.

So I propose we either:

1. Leave as-is, OR
2. Break the `Application` API and have `RegisterTendermintService` return an error.

Reply:
Ok for 1.